api/graphql/reservations/reservation_serializers.py

```diff
@@ -401,6 +401,9 @@ def validated_data(self):
         validated_data["state"] = STATE_CHOICES.CONFIRMED
         return validated_data
 
+    def check_metadata_fields(self, data, reservation_unit) -> None:
```

Comment:
What happens if the steps has been like: create (without metadata) => confirm ? Will it pass confirm - even if should not?

Reply:
Very good point! I didn't think of that scenario. 😖 In that scenario the confirmation would succeed happily, which is not what we want.

With that in mind, I think it's better to do the validation also in `confirmReservation`, but use the existing reservation data as the fallback. I believe that should cover that scenario.

Reply:
Fixed this and added a new test case to cover the scenario you described. 🙂